docs/sources/clients/promtail/configuration.md

```diff
@@ -1622,6 +1626,116 @@ users with thousands of services it can be more efficient to use the Consul API
 directly which has basic support for filtering nodes (currently by node
 metadata and a single tag).
 
+### docker_sd_config
+
+Docker service discovery configurations allow retrieving targets from a Docker daemon.
+It will only watch containers of the daemon referenced with the host parameter. That
+means it should run on each node in a distributed setup. The containers must run with
+either the [json-file](https://docs.docker.com/config/containers/logging/json-file/)
+or [journald](https://docs.docker.com/config/containers/logging/journald/) logging driver.
+
+Please note that the discovery will not pick up finished containers. That means
+Promtail will not scrape the remaining logs from finished containers after a restart.
+
+The configuration is inherited from [Prometheus' Docker service discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#docker_sd_config).
+
+```yaml
+# Address of the Docker daemon.  Use unix:///var/run/docker.sock for a local setup.
+host: <string>
+
+# Optional proxy URL.
+[ proxy_url: <string> ]
+
+# TLS configuration.
+tls_config:
+  [ <tls_config> ]
+
+# The port to scrape metrics from, when `role` is nodes, and for discovered
+# tasks and services that don't have published ports.
+[ port: <int> | default = 80 ]
+
+# The host to use if the container is in host networking mode.
+[ host_networking_host: <string> | default = "localhost" ]
+
+# Optional filters to limit the discovery process to a subset of available
+# resources.
+# The available filters are listed in the upstream documentation:
```

Comment:
Suggestion made because "upstream" and "downstream" are not well defined. 
```suggestion
# The available filters are listed in the Docker documentation:
```